src/components/Notifications/EventLog/usePrimaryToolbarFilterConfigWrapper.tsx

```diff
@@ -0,0 +1,180 @@
+import { ColumnsMetada, usePrimaryToolbarFilterConfig } from '@redhat-cloud-services/insights-common-typescript';
+import produce from 'immer';
+import React from 'react';
+import { useMemo } from 'react';
+
+import { Schemas } from '../../../generated/OpenapiNotifications';
+import { ClearEventLogFilters, EventLogFilterColumn, EventLogFilters, SetEventLogFilters } from './EventLogFilter';
+import { EventLogTreeFilter } from './EventLogTreeFilter';
+
+export interface EventLogCustomFilter {
+    bundleId: string,
+    category: string,
+    chips: Array<{ name: string, value: string, isRead: boolean }>
+}
+
+// Wrapper hook that gets the PrimaryToolbarFilterConfig and adds a custom conditional filter using Dropdown/Tree components
+// usePrimaryToolbarFilterConfig only supports 3 filter types: checkbox, radio, and text, so this extends that
+export const usePrimaryToolbarFilterConfigWrapper = (
+    bundles: readonly Schemas.Facet[],
+    filters: EventLogFilters,
+    setFilters: SetEventLogFilters,
+    clearFilter: ClearEventLogFilters,
+    metaData: ColumnsMetada<typeof EventLogFilterColumn>
+) => {
+    const [ customFilters, setCustomFilters ] = React.useState([] as EventLogCustomFilter[]);
+    const toolbarConfig = usePrimaryToolbarFilterConfig(
+        EventLogFilterColumn,
+        filters,
+        setFilters,
+        clearFilter,
+        metaData
+    );
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    const defaultDelete = React.useMemo(() => toolbarConfig.activeFiltersConfig.onDelete, []);
+    toolbarConfig.activeFiltersConfig.onDelete = React.useCallback((
+        _event: any,
+        rawFilterConfigs: any[]
+    ) => {
+        try {
+            defaultDelete(_event, rawFilterConfigs);
+        }
+        catch (e) {
+            setCustomFilters(produce(prev => {
+                const idxToRemove: number[] = [];
+                prev.forEach((activeFilter, idx) => {
+                    rawFilterConfigs.some(deleteFilter => {
+                        if (activeFilter.bundleId === deleteFilter.bundleId) {
+                            const deletedChipValues = deleteFilter.chips.map(chip => chip.value);
+                            activeFilter.chips = activeFilter.chips.filter(chip => !deletedChipValues.includes(chip.value));
+
+                            if (activeFilter.chips.length === 0) {
+                                idxToRemove.push(idx);
+                            }
+
+                            return true;
+                        }
+
+                        return false;
+                    });
+                });
+
+                idxToRemove.forEach((indexRemove, idx) => {
+                    prev.splice(indexRemove - idx, 1);
```

Comment:
Mutate inside forEach (tested in sandbox, forEach is smart enough to index the pre computed length of array and idx param also changes accordingly)